### DISCUSSION TITLE
Downsides of using the development server exclusively?

### DISCUSSION THREAD

**Discussion by Lab-Things** [mkdocs/mkdocs#2666] [Q&A]
Hi folks,

I was wondering whether there are any downside to using the mkdocs serve / development server exclusively instead of building/deploying. The documentation mentions this is not recommended, but doesn't really explain why not.

The only issue/downside I can think of is overhead, are there any other downsides? Security issues?

Kind regards,
Bob

**Comment by oprypin** [comment]
The thing is, there are literally no upsides. It's the same as running `mkdocs build` followed by `python -m http.server`. You can't say "instead of building", the files are still in the file system, although under a temporary path.

There's no configurability whatsoever either.

If you have any server at all, you would surely prefer to attach this to your existing server.

Security and performance were often named as downsides, and that may still apply (not as much as in the past though) but nobody has really checked it. Well, other than https://github.com/mkdocs/mkdocs/issues/2601 😳

**Reply by Lab-Things** [reply]
Thanks for the reply and also for mentioning #2601, for me personally the upsides are that I don't always have terminal access, but I do have file access. This way I know I'm always viewing the latest entries. Ofcourse there are also solutions to this, just exploring the playing field :)

**Reply by oprypin** [reply]
@Lab-Things Seems like that's not a "deployment" use case but rather indeed an intended "preview" use case for livereload - for that it's not surprising that it is helpful.
